src/stackWalker.cpp

```diff
@@ -379,7 +390,31 @@ int StackWalker::walkVM(void* ucontext, ASGCT_CallFrame* frames, int max_depth,
                 }
             }
         } else {
-            fillFrame(frames[depth++], BCI_NATIVE_FRAME, profiler->findNativeMethod(pc));
+            const char* method_name = profiler->findNativeMethod(pc);
+            char mark = method_name != NULL ? NativeFunc::mark(method_name) : 0;
+
+            if (java_pc_observed) {
+                fillFrame(frames[depth++], BCI_NATIVE_FRAME, method_name);
+            } else if (mark == MARK_ASYNC_PROFILER && event_type == MALLOC_SAMPLE) {
+                // Skip any frames above profiler hook methods
+                depth = 0;
+                fillFrame(frames[depth++], BCI_NATIVE_FRAME, method_name);
+            } else if (mark == MARK_COMPILER_ENTRY && Profiler::instance()->features().comp_task) {
```

Comment:
If `mark == MARK_COMPILER_ENTRY` and `Profiler::instance()->features().comp_task` is `false`, you are going to fall into the `else` statement and treat the frame as a normal native frame. Is that intended?